### ISSUE TITLE
dbt Cloud Ignoring Configuration File

### ISSUE THREAD

**Issue by denisefoster** [sqlfluff/sqlfluff#5751]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

While linting a model in dbt Cloud, SQL Fluff is configured with a customized file but is ignoring the excluded rules in the file and still flagging these as issues. Please see below.

![image](https://github.com/sqlfluff/sqlfluff/assets/130695571/90df14b3-3d78-45df-aa76-9cca9680c6b1)

### Expected Behaviour

When using Lint in dbt Cloud, I would expect to see the excluded_rules being ignored and not flagged as issues. When using Fix, code is formatted in a way that goes against our internal style.

### Observed Behaviour

Based on the logging, it looks like the .sqlfluff file is being ignored or not found. Please see screenshots below.

![image](https://github.com/sqlfluff/sqlfluff/assets/130695571/f8c612c6-fb45-4de7-a891-fd14f53039c3)

![image](https://github.com/sqlfluff/sqlfluff/assets/130695571/7aa08566-ac99-469c-8aa8-a8b4432b55a1)

![image](https://github.com/sqlfluff/sqlfluff/assets/130695571/26f60702-a071-4c6a-b9ff-8822e6fb61e4)

## .sqlfluff Config File

```
[sqlfluff]
verbose = 1
dialect = snowflake
templater = dbt
max_line_length = 200

exclude_rules = L034,layout.long_lines,aliasing.forbid,structure.column_order,ST06

# Turn off color formatting of output
nocolor = False
```

## SQL Code

```
SELECT

    --- ids
    order_id,

    --- strings
    administration_type,

    --- numerics
    COUNT(order_id) AS count_of_orders,

    --- booleans
    is_reorder,
    has_missed_tat,
    has_back_order_request,

    --- dates
    order_received_date,
    due_date,

    --- timestamps
    back_order_requested_at_utc,
    back_order_authorized_at_utc

FROM {{ ref('DA_STG_VIEW_ACHOO_US_ORDERS') }}

--- filter dates to show full past year through YTD and filter out future dates
WHERE YEAR(order_received_date) BETWEEN YEAR(CURRENT_DATE()) - 1 AND YEAR(CURRENT_DATE())
    AND CURRENT_DATE() > order_received_date

GROUP BY ALL
```

### Dialect

Snowflake

### Version

SQLFluff v3.0.3

dbt version 1.7.11

![image](https://github.com/sqlfluff/sqlfluff/assets/130695571/b7df983d-a9d7-4a17-915d-0b6b2cc1b954)

## .sqlfluff Config File

```
[sqlfluff]
verbose = 1
dialect = snowflake
templater = dbt
max_line_length = 200

exclude_rules = L034,layout.long_lines,aliasing.forbid,structure.column_order,ST06

# Turn off color formatting of output
nocolor = False
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by greg-mckeon** [comment]
hi @denisefoster - PM at dbt labs here. we had a recent issue where dbt Cloud was ignoring nested .sqlfluff config files. were you using multiple nested config files? if so, a fix for this should roll out next week.

either way, this one isn't on the SQLFluff team, so I think we can close!

**Comment by WittierDinosaur** [comment]
Thanks @greg-mckeon - closing on the SqlFluff side 👍 @denisefoster if useful, dbt have a public slack, and I've found them to generally be pretty responsive on it
